iceoryx_hoofs/container/include/iox/uninitialized_array.hpp

```diff
@@ -77,7 +77,7 @@ class UninitializedArray final
     using iterator = ElementType*;
     using const_iterator = const ElementType*;
 
-    constexpr UninitializedArray() noexcept = default;
+    constexpr UninitializedArray() noexcept {};
```

Comment:
It seems clang-tidy does not complain but please add a comment why we cannot use `= default` here :)

Reply:
@elBoberido Added. Additionally though, the added test would fail if someone tries to change it.